feather/protocol/src/io.rs

```diff
@@ -388,51 +392,57 @@ where
         let vec = iter::repeat_with(|| T::read(buffer, version))
             .take(length)
             .collect::<anyhow::Result<Vec<T>>>()?;
-        Ok(LengthPrefixedVec(Cow::Owned(vec)))
+        Ok(Self(Cow::Owned(vec), PhantomData))
     }
 }
 
-impl<'a, T> Writeable for LengthPrefixedVec<'a, T>
+impl<'a, P, T> Writeable for LengthPrefixedVec<'a, P, T>
 where
     T: Writeable,
     [T]: ToOwned<Owned = Vec<T>>,
+    P: TryInto<usize> + TryFrom<usize> + Writeable,
+    <P as TryFrom<usize>>::Error: std::fmt::Debug,
 {
     fn write(&self, buffer: &mut Vec<u8>, version: ProtocolVersion) {
-        VarInt::from(self.0.len()).write(buffer, version);
+        P::try_from(self.0.len()).unwrap().write(buffer, version);
```

Comment:
I didn't know how to handle this error, so I just unwrapped it. Will it cause problems?

Reply:
Hmm yeah not sure right now. Panicking should be avoided whenever necessary and reaching 65000 vec elements in case of a ShortPrefixedVec is not completely unrealistic.
Not sure what the best course of action would be

Reply:
I'm considering making the function return a result, but I don't really know if the error can be propagated in all instances.